lib/symbol-provider.coffee

```diff
@@ -210,58 +212,6 @@ class SymbolProvider
     suffix = lineFromPosition.match(@beginningOfLineWordRegex)?[0] or ''
     prefix + suffix
 
-  fuzzyFilter: (symbolList, buffer, {bufferPosition, prefix}) ->
-    # Probably inefficient to do a linear search
-    candidates = []
-
-    if @useAlternateScoring
-      fuzzaldrinProvider = fuzzaldrinPlus
-      # This allows to pre-compute and re-use some quantities derived from prefix such as
-      # Uppercase, lowercase and a version of prefix without optional characters.
-      prefixCache = fuzzaldrinPlus.prepQuery(prefix)
-    else
-      fuzzaldrinProvider = fuzzaldrin
-      prefixCache = null
-
-    for symbol in symbolList
-      text = (symbol.snippet or symbol.text)
-      continue unless text and prefix[0].toLowerCase() is text[0].toLowerCase() # must match the first char!
-      score = fuzzaldrinProvider.score(text, prefix, prefixCache)
-      if @useLocalityBonus then score *= @getLocalityScore(bufferPosition, symbol.bufferRowsForBuffer?(buffer))
-      candidates.push({symbol, score}) if score > 0
-
-    candidates.sort(@symbolSortReverseIterator)
-
-    results = []
-    for {symbol, score}, index in candidates
-      break if index is 20
-      results.push(symbol)
-    results
-
-  symbolSortReverseIterator: (a, b) -> b.score - a.score
-
-  getLocalityScore: (bufferPosition, bufferRowsContainingSymbol) ->
-    if bufferRowsContainingSymbol?
-      rowDifference = Number.MAX_VALUE
-      rowDifference = Math.min(rowDifference, bufferRow - bufferPosition.row) for bufferRow in bufferRowsContainingSymbol
-      locality = @computeLocalityModifier(rowDifference)
-      locality
-    else
-      1
-
-  computeLocalityModifier: (rowDifference) ->
-    rowDifference = Math.abs(rowDifference)
-    if @useAlternateScoring
-      # Between 1 and 1 + strength. (here between 1.0 and 2.0)
-      # Avoid a pow and a branching max.
-      # 25 is the number of row where the bonus is 3/4 faded away.
-      # strength is the factor in front of fade*fade. Here it is 1.0
-      fade = 25.0 / (25.0 + rowDifference)
-      1.0 + fade * fade
-    else
-      # Will be between 1 and ~2.75
-      1 + Math.max(-Math.pow(.2 * rowDifference - 3, 3) / 25 + .5, 0)
-
```

Comment:
Moving all this code from `SymbolProvider` to `SymbolStore` allows us to reduce the constant factor (of iterating over symbols) to `1`. This seems sounds also from a design perspective, because we treat `SymbolStore` as a database whose concern is to insert, search and score symbols.